packages/angular/cli/models/architect-command.ts

```diff
@@ -166,34 +166,32 @@ export abstract class ArchitectCommand extends Command<ArchitectCommandOptions>
       if (!targetSpec.project && this.target) {
         // This runs each target sequentially.
         // Running them in parallel would jumble the log messages.
-        return await from(this.getProjectNamesByTarget(this.target)).pipe(
+        return from(this.getProjectNamesByTarget(this.target)).pipe(
           concatMap(project => runSingleTarget({ ...targetSpec, project })),
           toArray(),
         ).toPromise().then(results => results.every(res => res === 0) ? 0 : 1);
       } else {
-        return await runSingleTarget(targetSpec).toPromise();
+        return runSingleTarget(targetSpec).toPromise();
       }
     } catch (e) {
       if (e instanceof schema.SchemaValidationException) {
         const newErrors: schema.SchemaValidatorError[] = [];
-        e.errors.forEach(schemaError => {
+        for (const schemaError of e.errors) {
           if (schemaError.keyword === 'additionalProperties') {
             const unknownProperty = schemaError.params.additionalProperty;
             if (unknownProperty in options) {
               const dashes = unknownProperty.length === 1 ? '-' : '--';
               this.logger.fatal(`Unknown option: '${dashes}${unknownProperty}'`);
-
-              return 1;
             }
           }
           newErrors.push(schemaError);
-        });
+        };
 
         if (newErrors.length > 0) {
           this.logger.error(new schema.SchemaValidationException(newErrors).message);
-
-          return 1;
         }
+
+        return 1;
```

Comment:
I'm pretty sure `1` here is the status code, which is error. You mean to do `return 0` here and leave the `return 1` above.

Reply:
it's inside the catch block.  If it got in there then there's an error.  The return was moved out of above so that all unknown options are reported instead of just the first.

Reply:
reverted behavioral change.